server/logging/firehose.go

```diff
@@ -49,6 +50,17 @@ func NewFirehoseLogWriter(region, id, secret, stream string, logger log.Logger)
 	if err != nil {
 		return nil, errors.Wrap(err, "create Firehose client")
 	}
+
+	if stsAssumeRoleArn != "" {
```

Comment:
Can we refactor this a bit so that there is only a single call to `session.NewSession` and above that is something like
```
if stsAssumeRoleArn != "" {
    creds := stscreds.NewCredentials(sess, stsAssumeRoleArn)
    conf.Credentials = creds
}
```

Reply:
As far as I can tell from reading docs about aws-sdk-go, I don't think this is possible, but I'm not super familiar with it.

AFAICT you need to have an existing AWS session to get sts credentials.

Reply:
Apologies for the delay. What I was referring to was using a strategy like on lines 45-57.

Something like this on line 48

```
if stsAssumeRoleArn != "" {
    creds := stscreds.NewCredentials(sess, stsAssumeRoleArn)
    conf.Credentials = creds
}
```

Then we have only one place where `sess, err = session.NewSession(conf)` appears.

Reply:
I missed that creating the credentials requires an already active session. This looks good.